xml/System.Text.Json/JsonSerializer.xml

```diff
@@ -385,12 +447,50 @@ A value could not be read from the reader.</exception>
         <Parameter Name="context" Type="System.Text.Json.Serialization.JsonSerializerContext" Index="2" FrameworkAlternate="net-6.0" />
       </Parameters>
       <Docs>
-        <param name="reader">To be added.</param>
-        <param name="returnType">To be added.</param>
-        <param name="context">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="reader">The reader to read.</param>
+        <param name="returnType">The type of the object to convert to and return.</param>
+        <param name="context">A metadata provider for serializable types.</param>
+        <summary>Reads one JSON value (including objects or arrays) from the provided reader into a <paramref name="returnType" />.</summary>
+        <returns>A <paramref name="returnType" /> representation of the JSON value.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+
+                 If the <xref:System.Text.Json.Utf8JsonReader.TokenType> property of `reader`
+                 is <xref:System.Text.Json.JsonTokenType.PropertyName> or <xref:System.Text.Json.JsonTokenType.None>, the
+                 reader will be advanced by one call to <xref:System.Text.Json.Utf8JsonReader.Read> to determine
+                 the start of the value.
+               
+
+
+                 Upon completion of this method `reader` will be positioned at the
+                 final token in the JSON value.  If an exception is thrown the reader is reset to
+                 the state it was in when the method was called.
+               
+
+
+                 This method makes a copy of the data the reader acted on, so there is no caller
+                 requirement to maintain data integrity beyond the return of this method.
+               
+
+
+                 The <xref:System.Text.Json.JsonReaderOptions> used to create the instance of the <xref:System.Text.Json.Utf8JsonReader> take precedence over the <xref:System.Text.Json.JsonSerializerOptions> when they conflict.
+                 Hence, <xref:System.Text.Json.JsonReaderOptions.AllowTrailingCommas>, <xref:System.Text.Json.JsonReaderOptions.MaxDepth>, and <xref:System.Text.Json.JsonReaderOptions.CommentHandling> are used while reading.
+               
+
+
+
+          ]]></format>
+        </remarks>
+        <exception cref="T:System.ArgumentNullException">
+          <paramref name="returnType" /> or <paramref name="context" /> is <see langword="null" />.</exception>
+        <exception cref="T:System.Text.Json.JsonException">Thrown when the JSON is invalid, <paramref name="returnType" /> is not compatible with the JSON, or a value could not be read from the reader.</exception>
```

Comment:
```suggestion
        <exception cref="T:System.Text.Json.JsonException">The JSON is invalid, <paramref name="returnType" /> is not compatible with the JSON, or a value could not be read from the reader.</exception>
```

Reply:
I see the code is full of this pattern, eg., "/// Thrown if this would result in invalid JSON being written (while validation is enabled)." 

To make code the source of truth, we will either have to train devs on these patterns, or relax our expectations. I suspect a mixture. The docs will no longer have a 'single voice' but will hopefully be more complete and up to date.